### PR TITLE
Test MODEL_TYPE without SparkSession

### PR DESCRIPTION
**Pull request by da-tubi**

### CODE REVIEW

File: python/rikai/spark/sql/codegen/pytorch.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def move_tensor_to_device(data, device):
     return data
 
 
-def generate_udf(payload: ModelSpec):
+def generate(payload: ModelSpec, is_udf: bool = True):
```
Comment (changhiskhan) [inline python/rikai/spark/sql/codegen/pytorch.py:44]: `generate_inference_func` ?

File: python/rikai/spark/sql/codegen/base.py [hunk 2]
```diff
@@ -67,28 +98,19 @@ def udf_from_spec(spec: ModelSpec):
...
 
-    try:
-        codegen = importlib.import_module(codegen_module)
-        return (
-            pickle_udt,
-            codegen.generate_udf(spec),
-            deserialize_return,
-            schema,
-        )
-    except ModuleNotFoundError:
```
Comment (changhiskhan) [inline python/rikai/spark/sql/codegen/base.py:89]: is it a good idea to just let random model flavors in?

File: experimental/torchhub/rikai/experimental/torchhub/torchhub_registry.py [hunk 3]
```diff
@@ -27,11 +27,11 @@ class TorchHubModelSpec(ModelSpec):
     def __init__(self, repo_or_dir: str, model: str, raw_spec: "ModelSpec"):
         spec = {
             "version": "1.0",
-            "schema": raw_spec["schema"],
+            "schema": raw_spec.get("schema", None),
```
[the remark this answers is not in the document]
Comment (darcy-shen) [reply experimental/torchhub/rikai/experimental/torchhub/torchhub_registry.py:30]: Yes. Using model type, scheme is in Model Type

[the remark this answers is not in the document]
Comment (darcy-shen) [reply experimental/torchhub/rikai/experimental/torchhub/torchhub_registry.py:30]: See https://github.com/eto-ai/rikai/blob/c80c2c23df30fd2a1c71df8829e1148d77878f71/python/rikai/pytorch/models/torchvision.py#L46-L47

File: python/rikai/spark/sql/codegen/base.py (hunk 2 above)
Comment (darcy-shen) [reply python/rikai/spark/sql/codegen/base.py:89]: no changes for this part, these deleted code is refactored in `codegen_from_spec` and is reused in `func_from_spec`

File: experimental/torchhub/tests/test_resnet.py [hunk 4]
```diff
@@ -13,18 +13,37 @@
...
 from rikai.spark.functions import to_image
+from rikai.testing.utils import apply_model_spec
+
+
+version = f"v{torchvision.__version__.split('+', maxsplit=1)[0]}"
+work_dir = Path().absolute().parent.parent
+image_path = f"{work_dir}/python/tests/assets/test_image.jpg"
+
+
+def test_resnet_model_type():
```
Comment (da-tubi) [inline experimental/torchhub/tests/test_resnet.py:31]: unit test without spark session case 1

Comment (darcy-shen) [inline experimental/torchhub/tests/test_resnet.py:31]: unit test without spark session case 1

File: python/tests/spark/sql/codegen/test_fs.py [hunk 5]
```diff
@@ -193,6 +195,21 @@ def test_yaml_model(spark: SparkSession, resnet_spec: str):
     check_ml_predict(spark, "resnet_m")
 
 
+def test_yaml_model_type(resnet_spec: str):
```
Comment (darcy-shen) [inline python/tests/spark/sql/codegen/test_fs.py:198]: unit test without spark session case 2

**Review by Renkai — APPROVED** [review]

**Pull request merged**